.github/workflows/tsp-client-tests.yml

```
@@ -0,0 +1,55 @@
+name: tsp-client - Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - tools/tsp-client/**
```

```suggestion
      - .github/workflows/tsp-client-tests.yml
      - tools/tsp-client/**
```

You want to also trigger the action if the workflow YML itself is updated.  This change should also run your new action on this very PR.